lib/psi.rb

```diff
@@ -61,79 +64,256 @@ def append(k)
+      balcony:       0.500, # *
+      party:         0.450, # ** "regular (BETBG)"
+      grade:         0.450, # *
+      joint:         0.200, # ** "regular (BETBG)"
+      transition:    0.000
+    }.freeze               # based on EXTERIOR dimensions (art. 3.1.1.6)
+    self.genShorthands("code (Quebec)")
+
+    @set["(non thermal bridging)"] = # ... would not derate surfaces:
     {
-      rimjoist:     0.000, #
-      parapet:      0.000, #
-      fenestration: 0.000, #
-      concave:      0.000, #
-      convex:       0.000, #
-      balcony:      0.000, #
-      party:        0.000, #
-      grade:        0.000  #
+      rimjoist:      0.000,
+      parapet:       0.000,
+      fenestration:  0.000,
+      corner:        0.000,
+      balcony:       0.000,
+      party:         0.000,
+      grade:         0.000,
+      joint:         0.000,
+      transition:    0.000
     }.freeze
+    self.genShorthands("(non thermal bridging)")
+  end
+
+  ##
+  # Generate PSI set shorthand listings
+  # Requires a valid, unique :id.
+  #
+  # @param [String] p A PSI set identifier
+  def genShorthands(p)
+    if @set.has_key?(p)
+      h = {}
+      h[:joint]           = @set[p].has_key?(:joint)
+      h[:transition]      = @set[p].has_key?(:transition)
+      h[:fenestration]    = @set[p].has_key?(:fenestration)
+      h[:head]            = @set[p].has_key?(:head)
+      h[:headconcave]     = @set[p].has_key?(:headconcave)
+      h[:headconvex]      = @set[p].has_key?(:headconvex)
+      h[:sill]            = @set[p].has_key?(:sill)
+      h[:sillconcave]     = @set[p].has_key?(:sillconcave)
+      h[:sillconvex]      = @set[p].has_key?(:sillconvex)
+      h[:jamb]            = @set[p].has_key?(:jamb)
+      h[:jambconcave]     = @set[p].has_key?(:jambconcave)
+      h[:jambconvex]      = @set[p].has_key?(:jambconvex)
+      h[:corner]          = @set[p].has_key?(:corner)
+      h[:cornerconcave]   = @set[p].has_key?(:cornerconcave)
+      h[:cornerconvex]    = @set[p].has_key?(:cornerconvex)
+      h[:parapet]         = @set[p].has_key?(:parapet)
+      h[:partyconcave]    = @set[p].has_key?(:parapetconcave)
+      h[:parapetconvex]   = @set[p].has_key?(:parapetconvex)
+      h[:party]           = @set[p].has_key?(:party)
+      h[:partyconcave]    = @set[p].has_key?(:partyconcave)
+      h[:partyconvex]     = @set[p].has_key?(:partyconvex)
+      h[:grade]           = @set[p].has_key?(:grade)
+      h[:gradeconcave]    = @set[p].has_key?(:gradeconcave)
+      h[:gradeconvex]     = @set[p].has_key?(:gradeconvex)
+      h[:balcony]         = @set[p].has_key?(:balcony)
+      h[:balconyconcave]  = @set[p].has_key?(:balconyconcave)
+      h[:balconyconvex]   = @set[p].has_key?(:balconyconvex)
+      h[:rimjoist]        = @set[p].has_key?(:rimjoist)
+      h[:rimjoistconcave] = @set[p].has_key?(:rimjoistconcave)
+      h[:rimjoistconvex]  = @set[p].has_key?(:rimjoistconvex)
+      @has[p] = h
+
```

Comment:
"@has" : True/False response as to whether a PSI set contains a given thermal bridge type. Relying on Topolys, TBD tags an edge with one (or more) of the above labels. This is systematically populated once a PSI set is added and held in memory for quick access.